### DISCUSSION TITLE
"Extract into function" puts extra "\" in generated function only in WSL

### DISCUSSION THREAD

**Discussion by elithrade** [rust-lang/rust-analyzer#16977] [Q&A]
Hi,

I have a strange problem I cannot figure out why.
I have the following code:
```
fn main() {
    let n = 100;
    for i in 0..n {
        println!("{}", i);
    }
}
```
I select the for loop and choose "Extract into function" the generated function looks like this:
```
fn main() {
    let n = 100;
    fun_name(n);
}

fn fun_name(n: i32) {
    for i in 0..n {
        println!("{\}", i);
    \}
\}
```
![image](https://github.com/rust-lang/rust-analyzer/assets/10831354/09de99a0-751d-40c0-8f09-3b8b88c22ae0)

I cannot figure out where those "\" come from. I am using Neovim in WSL. My rist-anaylzer version is:
❯ rust-analyzer --version
rust-analyzer 1.77.1 (7cf61eb 2024-03-27)

I have the same setup on my Windows machine and it's working as expected. Any ideas? Is this a WSL issue? Any helps appreciated.

**Comment by lnicola** [comment]
You probably have a different version on Windows. Can you test with a binary release from GitHub?

**Reply by elithrade** [reply]
I tried the latest binary, still getting the same problem,
![image](https://github.com/rust-lang/rust-analyzer/assets/10831354/577f0b3e-59d4-4c96-99d8-f7d1e67c34d4)

My Windows version is
![image](https://github.com/rust-lang/rust-analyzer/assets/10831354/9683dac8-78c1-4302-bd11-c79cce96d5e5)

**Comment by elithrade** [comment]
https://github.com/rust-lang/rust-analyzer/issues/16607 sounds like the same issue.

**Reply by lnicola** [reply]
Yeah, but that one's fixed.

CC @DropDemBits

**Reply by DropDemBits** [reply]
It's actually #16867 which is an issue on Neovim's side. It's fixed in v0.10 (which looks like it'll be released in May), though the workaround described in https://github.com/rust-lang/rust-analyzer/issues/16867#issuecomment-2009391788 works too:

> As a temporary workaround, if you're using `rustaceanvim`, you can set `vim.g.rustaceanvim.server.capabilities.experimental.snippetTextEdit` to `false` (https://github.com/mrcjkb/rustaceanvim/issues/303#issuecomment-2009381608).
If you're using something else, you should be able to tweak the flag in a similar way in Lua.
